doc/rfcs/002-update-buffer-titles-in-guest.md

```diff
@@ -0,0 +1,48 @@
+# Update Buffer Titles in Guest's Workspace
+
+### Status
+
+Proposed
+
+### Summary
+
+We want to add the functionality to have title changes of text-buffers from the host using Teletype to be reflected in the guests' buffers.
+
+### Motivation
+
+As described in [#147](https://github.com/atom/teletype/issues/147), when the host renames a file, the guest should see those changes reflected in their workspace as well.
+
+### Explanation
+
+This will add an additional workflow to the _Teletype_ process:
+
+1. A subscription is added to _teletype/Buffer-Binding_ to capture when the _Text-Buffer_'s path changes. This will then get what the new URI of the host's _teletype-client/Buffer-Proxy_ should be, and call a function in _Buffer-Proxy_.
+2. The Host's _Buffer-Proxy_ will call router to notify all of the guests the new URI of the buffer. After this notification, it will update its URI.
+3. The Guest's _Buffer-Proxy_, will add a subscription (upon initialization) to listen for path change notifications from router.
+4. Upon router's notification and it will update its URI. It will then notify the _Text-Buffer_ to send an emitter that its path has changed.
+5. This emitter is called by an added helper function in _atom/Text-Buffer_.
+6. The Guest's _teletype/Editor-Binding_'s monkey bindings are updated such that the URI constant is removed, and is updated when `getTitle()` is invoked.
```

Comment:
I think Step 5 and 6 will change slightly if you follow the suggestion described in https://github.com/atom/text-buffer/pull/292#issuecomment-371741409. Does that sound right?

Reply:
That sounds right, we'll make a revision once we're sure which avenue we'll follow on it.